starlite/contrib/jwt/jwt_token.py

```diff
@@ -0,0 +1,127 @@
+from datetime import datetime, timezone
+from typing import Dict, Optional, Union, cast
+
+from jose import JWSError, JWTError, jwt
+from pydantic import (
+    BaseConfig,
+    BaseModel,
+    Extra,
+    Field,
+    ValidationError,
+    constr,
+    validator,
+)
+
+from starlite import ImproperlyConfiguredException
+from starlite.exceptions import NotAuthorizedException
+
+
+def _normalize_datetime(value: datetime) -> datetime:
+    """Convert the given value into UTC and strips microseconds.
+
+    Args:
+        value: A datetime instance
+
+    Returns:
+        A datetime instance
+    """
+    if value.tzinfo is not None:
+        value.astimezone(timezone.utc)
+    return value.replace(microsecond=0)
+
+
+class Token(BaseModel):
```

Comment:
Does this really need to be a pydantic model?
I feel like since we plan to drop pydantic as a hard dependency, we shouldn't unnecessarily be adding new pydantic-dependant functionality, since we'll need to remove it at some point not too far in the future anyway.

(This applies to other pydantic usage here as well)

Reply:
i will refactor, now that tests are there.

Reply:
So after spending some time on this - yes, we should use pydantic here. Its way simpler and cleaner. We can also not drop pydantic as a dependency because we use it for OpenAPI schema generation, so there is no point in fighting this.

Reply:
So this is off the table then?

> One change I want to discuss is to switch from Pydantic to dataclasses as the default fir Starlite. I don't mean remove Pydantic at all, but rather make Pydantic an optional dependency, without which Starlite will default to using the stdlib dataclasses.

The conclusion you drew here is quite the setback then in regard to supporting other serialization/validation libraries. If we still need to use pydantic under the hood to generate OpenAPI schemas, it severely limits the scope of where we can actually drop pydantic.

Also, it may be simpler and cleaner with pydantic, but I think it will harm performance substantially. If the token is a pydantic model which is being validated on every request sent, it could practically halve the performance gain in some situations, if someone were to pick something else for serialization.

I feel like *if* we make it possible to opt for the serialization/validation library of your choice, we shouldn't be sneaking pydantic in elsewhere. Using it for something like configs and stuff I can get behind, as that's a cost only paid at startup time. But what you're suggesting is that everything that has to be added to the OpenAPI schema **needs** to be pydantic, which means there are barely any places we can truly switch from pydantic.

I'd rather we invest the time and effort to somehow bridge these things, so they are not required to be pydantic models themselves. Because otherwise, we might as well not do the migration away from pydantic to begin with.

Reply:
Not really. It's a very large amount of work to remove Pydantic from our OpenAPI logic, and I don't see why we should do this. 

We can support different serialization / validation backends, this doesn't require we remove Pydantic. The effort required to remove this dependency is excessive and the benefit is too low. I mean, if you feel like putting in the 100+ hours we'd need for this change just so we have one less dependecy, go ahead, but I think it's not worth the effort. 

Finally - for config classes that are evaluated at startup, Pydantic is actually perfect. It's performance is not an issue in this case and it offers a vastly better alternative to dataclasses.

Reply:
> We can support different serialization / validation backends, this doesn't require we remove Pydantic. 
No, but we talked about removing it from the internal datastructures where it's not strictly necessary. 

> I mean, if you feel like putting in the 100+ hours we'd need for this change just so we have one less dependecy, go ahead

Not really, no :upside_down_face: 

> Finally - for config classes that are evaluated at startup, Pydantic is actually perfect. It's performance is not an issue in this case and it offers a vastly better alternative to dataclasses.

Yes, I agree with you there!

> > Using it for something like configs and stuff I can get behind